### PR TITLE
Simulator Cleanup

### PR DESCRIPTION
**Pull request by samliok**
Addressing a couple comments from the Simulator PR
- remove return values from insert + remove callbacks
- gets rid of `SimulatorState` struct in go since that was just a wrapper for `Mutable`
- some other small comments

### CODE REVIEW

File: x/programs/simulator/common/types.h [hunk 1]
```diff
@@ -57,9 +57,13 @@ typedef char *(*InsertStateCallback)(void *data, Bytes key, Bytes value);
 typedef char *(*RemoveStateCallback)(void *data, Bytes key);
 
 typedef struct {
-    // this ptr gives context to the callbacks
+    // this is a pointer to the state passed in from rust
+	// it points to the state object on the rust side of the simulator
```
Comment (samliok) [inline x/programs/simulator/common/types.h:61]: this is properly formatted on my IDE and when I download the file as well

Comment (richardpringle) [reply x/programs/simulator/common/types.h:61]: It's a hard tab instead of spaces

Comment (richardpringle) [inline x/programs/simulator/common/types.h:61]:
```suggestion
    // this is a pointer to the state passed in from rust
    // it points to the state object on the rust side of the simulator
```

File: x/programs/simulator/common/types.h [hunk 2]
```diff
@@ -57,9 +57,13 @@ typedef char *(*InsertStateCallback)(void *data, Bytes key, Bytes value);
...
     void *stateObj;
+	// this is ptr to the get function
```
Comment (richardpringle) [inline x/programs/simulator/common/types.h:63]:
```suggestion
    // this is ptr to the get function
```

File: x/programs/simulator/common/types.h [hunk 3]
```diff
@@ -57,9 +57,13 @@ typedef char *(*InsertStateCallback)(void *data, Bytes key, Bytes value);
...
     GetStateCallback get_value_callback;
+    // this is ptr to the insert function
     InsertStateCallback insert_callback;
+	// this is a ptr to the delete function
```
Comment (richardpringle) [inline x/programs/simulator/common/types.h:67]:
```suggestion
    // this is a ptr to the delete function
```

File: x/programs/simulator/src/lib.rs [hunk 4]
```diff
@@ -16,6 +16,9 @@ mod bindings {
         type Target = [u8];
 
         fn deref(&self) -> &Self::Target {
+            // # Safety:
+            // These bytes were mallocated by CGo
```
Comment (richardpringle) [inline x/programs/simulator/src/lib.rs:20]:
```suggestion
            // These bytes were allocated by CGo
```
"allocated" is the more generic term

File: x/programs/simulator/src/lib.rs [hunk 5]
```diff
@@ -16,6 +16,9 @@ mod bindings {
...
+            // They are guaranteed to be valid since for the length of the slice
```
Comment (richardpringle) [inline x/programs/simulator/src/lib.rs:21]:
```suggestion
            // They are guaranteed to be valid for the length of the slice
```

File: x/programs/simulator/state/manager.go [hunk 6]
```diff
@@ -18,9 +18,14 @@ import (
...
 )
 
-// ensure SimulatorStateManager implements StateManager
 var _ runtime.StateManager = &ProgramStateManager{}
 
+var (
+	ErrUnknownAccount = errors.New("unknown account")
+	balanceKeyBytes   = []byte("balance")
+	programKeyBytes   = []byte("program")
+)
```
Comment (richardpringle) [inline x/programs/simulator/state/manager.go:27]: 🥇

**Review by richardpringle — CHANGES_REQUESTED** [review]

**Review by richardpringle — APPROVED** [review]
🚀

**Pull request merged**